Pokemon/Pokemon/PKMasterTableViewController.m

```diff
@@ -0,0 +1,647 @@
+    
+    NSArray *steel = @[
+                       @"magnemite",
+                       @"magneton"
+                       ];
+    
+    
+    NSArray *water = @[
+                       @"blastoise",
+                       @"cloyster",
+                       @"dewgong",
+                       @"goldeen",
+                       @"golduck",
+                       @"gyarados",
+                       @"horsea",
+                       @"kabuto",
+                       @"kabutops",
+                       @"kingler",
+                       @"krabby",
+                       @"lapras",
+                       @"magikarp",
+                       @"omanyte",
+                       @"omastar",
+                       @"poliwag",
+                       @"poliwhirl",
+                       @"poliwrath",
+                       @"psyduck",
+                       @"seadra",
+                       @"seaking",
+                       @"seel",
+                       @"shellder",
+                       @"slowbro",
+                       @"slowpoke",
+                       @"squirtle",
+                       @"starmie",
+                       @"staryu",
+                       @"tentacool",
+                       @"tentacruel",
+                       @"vaporeon",
+                       @"wartortle"
+                       
+                       ];
+    
+    self.typePokemon = @[@{@"name":@"bug",
+                           @"list":bug},
+                         @{@"name":@"dragon",
+                           @"list":dragon},
+                         @{@"name":@"electric",
+                           @"list":electric},
+                         @{@"name":@"fairy",
+                           @"list":fairy},
+                         @{@"name":@"fire",
+                           @"list":fire},
+                         @{@"name":@"fighting",
+                           @"list":fighting},
+                         @{@"name":@"flying",
+                           @"list":flying},
+                         @{@"name":@"ghost",
+                           @"list":ghost},
+                         @{@"name":@"grass",
+                           @"list":grass},
+                         @{@"name":@"ice",
+                           @"list":ice},
+                         @{@"name":@"normal",
+                           @"list":normal},
+                         @{@"name":@"poison",
+                           @"list":poison},
+                         @{@"name":@"psychic",
+                           @"list":psychic},
+                         @{@"name":@"rock",
+                           @"list":rock},
+                         @{@"name":@"steel",
+                           @"list":steel},
+                         @{@"name":@"water",
+                           @"list":water},
+                         @{@"name":@"ground",
+                           @"list":ground}];
+    
+
+    
+    self.alphabeticalPokemon = @[@{@"name":@"A",
```

Comment:
Be careful when repeating data, a good principle to follow is [DRY(Don't repeat yourself)] (https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).  The problem is if you added a change or add a Pokemon for instance, you'd have to do it in two spots.  This makes things complicated and prone to errors as a program grows larger.

There are a few alternative ways to handle this.

A good way would be to have some abstract class for each Pokemon, which includes their name and type, and then another class something like PokemonList that includes an array of Pokemon.  You then could populate the PokemonList array with each Pokemon once.  You could then create methods in the PokemonList class that return arrays of Pokemon by type or all of them alphabetically.